sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/maskExpressions.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.commons.codec.digest.DigestUtils
+
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(str) - Returns hash of the given value. The hash is consistent and can be used to join masked values together across tables",
```

Comment:
then it's probably better to name this function `sha` to match `md5` and follow snowflake: https://docs.snowflake.com/en/sql-reference/functions/sha2.html

Reply:
Spark SQL  already has a built-in function [sha2](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L725)
This current PR is a specific implementation of sha2 with  bit length 512 or 256.  The intention of new new function **mask_hash** is to support a family of masking functions tracked through this JIRA

https://issues.apache.org/jira/browse/SPARK-40686
- mask
- mask_first_n(string str[, int n])
- mask_last_n(string str[, int n]
- mask_show_first_n(string str[, int n])
- mask_show_last_n(string str[, int n])
- **mask_hash**

Reply:
@vinodkc that explanation is helpful to understand the difference, thank you!

Reply:
Oh I see, shall we always use sha 512 to simplify the design? I'm not sure when people need to tune it.

Reply:
these are named from [Hive](https://cwiki.apache.org/confluence/display/hive/languagemanual+udf#LanguageManualUDF-DataMaskingFunctions).

Reply:
Since we already have `sha2`, this PR is just adding an alias. We need to prove that this alias is common in the industry. Except for Hive, do other systems/products use these names?